### DISCUSSION TITLE
What happens to the contents of the right-dock/assistant buffer when you press "new"?

### DISCUSSION THREAD

**Discussion by mattgiles** [zed-industries/zed#10191] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I was working on a relatively long-winded system prompt in the right dock.

I pressed "new". I think I lost my work as a result? Unclear where it can be found if not.

And like whatever my bad. But it really does feel like the kind of thing which should just not happen so easily in a text editor application. In a normal text buffer, for example, pressing new would not destroy the current buffer, but open a new tab.

It would be great to retain access to the old assistant buffer contents after pressing "new", maybe in some notion of history, or preserving it in a tab. Not sure.

**Comment by Moshyfawn** [comment]
The assistant conversation is stored in the history. You can access your history by pressing the "burger" button next to the conversation title.

<img width="230" alt="image" src="https://github.com/zed-industries/zed/assets/16290753/860168af-20cb-4a6e-b0b1-cf5e26056a54">

**Comment by mattgiles** [comment]
@Moshyfawn thanks for the response. I did click on the history burger immediately after losing my work. But in my case the history was a blank tile -- potentially because I had not yet run the prompt against chatGPT? I don't know what crosses the threshold for getting saved into conversation history, but whatever it is is implicit and my work didn't qualify.

I think my expectation putting words into a text editor -- even in a secondary, specialized buffer for LLM interaction -- is analogous to my expectations of my bank: when I give the bank dollars, I expect the bank to hold on to them.

**Reply by JosephTLyons** [reply]
Yeah, our assistant is pretty underdeveloped, but it is something we are going to be working on. It is currently in pretty much the same state as the day we introduced it. Right now, I think you are right - we only serialize the data to file after you launch your first request to the model, since the summary of the convo returned by the model is ultimately what we use for the file name when saving the contents. We should probably save the files temporarily until then.

Sorry, its a bit annoying that this under-developed part of Zed can cause lose of work - we will improve upon it and polish it up, as part of our overall AI effort within Zed.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
